_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+
+
+## How will I ask for this information?
+
+Will you design new custom input fields or use recognizable, familiar patterns? Please don’t reinvent the web for something as essential as form fields. As Heydon Pickering explains in his talk [Get Your Priorities Straight](https://www.youtube.com/watch?v=ediHVy0869c): “Real people aren't looking to be delighted. People want to get the task done and get on with their lives".
+
+{: .example }
+For example: users recognize radio buttons as circles and checkboxes as squares. They know from experience they can choose one option from the radio buttons and multiple options from the checkboxes. Also, the [keyboard interaction](https://webaim.org/techniques/keyboard/#testing) for radiobuttons and checkboxes differs from each other. 
+
+Some field types are better avoided due to poor usability, such as multi-selects. Using date pickers? Check the experience for keyboard- or screen reader users.
+
+Sometimes a simple solution works best for everyone — for example, KLM’s method for selecting a date with a text input for the day, a dropdown for the month, and another text input for the year:
+![Date selection with a text field for day, a select for month, and a text field for year](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/blog_toegankelijker-foutmeldingen-date-picker.png)
+
+For a deep dive into asking for dates in forms, see the GOV.UK Design System pattern library: [Ask users for Dates](https://design-system.service.gov.uk/patterns/dates/).
+
+Don't force the user to enter information twice (like an email address) and remember in a multistep form entries given in previous steps.
+
+{: .note }
+This success criterion is added in [WCAG 2.2: 3.3.7 Redundant Entry](https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry.html).
+
+## What information can I provide in advance to help the user?
+
+We’ve all been there: you enter a new password, click submit, and *then* you’re told what the password requirements are. You yell at the screen: “Tell me that beforehand!”.
+
+The blog post [*Blind people don't visit my website*](https://www.a11y-collective.com/blog/blind-people-dont-visit-my-website/) lists frustrations from web users, such as:
+*"Error messages that come afterwards and eventually don't explain what you need to change."*
+
+Don’t make users guess — offer clear help in plain language, at the right moment. Tell users in advance which documents they'll need, and which fields are or aren't required. Add descriptions to fields where necessary.
+
+## How to show which fields are required?
+
+If you only ask what's really needed, all fields could technically be required — but that’s not always the case.
```

Comment:
Maybe a good example could be added here. For example, if you're creating content in the admin of a website, all of the content fields you might use need to be available; but most of them won't be required.